Mathlib/Analysis/Calculus/Taylor.lean

```diff
@@ -209,6 +212,85 @@ theorem hasDerivWithinAt_taylorWithinEval_at_Icc {f : ℝ → E} {a b t : ℝ} (
   hasDerivWithinAt_taylorWithinEval (uniqueDiffOn_Icc hx t ht) (uniqueDiffOn_Icc hx)
     self_mem_nhdsWithin ht rfl.subset hf (hf' t ht)
 
+/-- Calculate the derivative of the Taylor polynomial with respect to `x`. --/
+theorem hasDerivAt_taylorWithinEval_succ {x₀ x : ℝ} {s : Set ℝ}
+    (hs : UniqueDiffOn ℝ s) (hx₀ : x₀ ∈ s) (f : ℝ → E) (n : ℕ) :
+    HasDerivAt (taylorWithinEval f (n + 1) s x₀)
+      (taylorWithinEval (derivWithin f s) n s x₀ x) x := by
+  change HasDerivAt (fun x ↦ taylorWithinEval f _ s x₀ x) _ _
+  simp_rw [taylor_within_apply]
+  have : ∀ (i : ℕ) {c : ℝ} {c' : E},
+      HasDerivAt (fun x ↦ (c * (x - x₀) ^ i) • c') ((c * (i * (x - x₀) ^ (i - 1) * 1)) • c') x :=
+    fun _ _ ↦ hasDerivAt_id _ |>.sub_const _ |>.pow _ |>.const_mul _ |>.smul_const _
+  apply HasDerivAt.sum (fun i _ => this i) |>.congr_deriv
+  rw [Finset.sum_range_succ', Nat.cast_zero, zero_mul, zero_mul, mul_zero, zero_smul, add_zero]
+  apply Finset.sum_congr rfl
+  intro i _
+  rw [← iteratedDerivWithin_succ' hs hx₀]
+  congr 1
+  field_simp [Nat.factorial_succ]
+  ring
+
+/-- **Taylor's theorem** as a limit. -/
+theorem taylor_tendsto {f : ℝ → ℝ} {x₀ : ℝ} {n : ℕ} {s : Set ℝ}
```

Comment:
You are stating the theorem for functions `f : ℝ → ℝ`, but the same statement (with essentially the same proof) should hold for `f : ℝ → E` (in the statement, instead of dividing by `(x - x_0)^n`, you should just use the scalar action on `E`. If you want to have a specific version for `E = ℝ`, why not, but I think the main version should be the general one.